src/inline_lazy.rs

```diff
@@ -40,6 +43,18 @@ unsafe impl<T: Sync> Sync for Lazy<T> {}
 #[doc(hidden)]
 macro_rules! __lazy_static_create {
     ($NAME:ident, $T:ty) => {
-        static mut $NAME: $crate::lazy::Lazy<$T> = $crate::lazy::Lazy(None, $crate::lazy::ONCE_INIT);
-    }
+        static mut $NAME: $crate::lazy::Lazy<$T> = $crate::lazy::Lazy::INIT;
+    };
+}
+
+#[cfg(lazy_static_core_unreachable_unchecked)]
+use core::hint::unreachable_unchecked;
```

Comment:
Do we need to bother with `core::hint::unreachable_unchecked` at all with our polyfill?

Reply:
Nope, we can wait until the minimum version is raised for some other reason, to switch to the `std` version.